packages/core/src/checker/checker-facade.ts

```diff
@@ -34,73 +35,67 @@ export class CheckerFacade {
     checkResult$: Observable<MutantResult>;
     passedMutant$: Observable<MutantRunPlan>;
   } {
-    if (!this.options.checkers.length) {
-      return {
-        checkResult$: EMPTY,
-        passedMutant$: input$,
-      };
-    }
-
-    const checkResult$ = new Subject<MutantResult>();
-    let previousPassedMutants$ = input$;
-
-    for (let i = 0; i < this.options.checkers.length; i++) {
-      const passedMutants$ = new Subject<MutantRunPlan>();
-      this.executeChecker(i, previousPassedMutants$, checkResult$, passedMutants$);
-      previousPassedMutants$ = passedMutants$;
+    let checkResult$: Observable<MutantResult> = EMPTY;
+    let passedMutant$ = input$;
+    for (const checkerName of this.options.checkers) {
+      // Use this checker
+      const { failedCheckResult$, passedCheckResult$ } = this.executeChecker(checkerName, passedMutant$);
+
+      // Prepare for the next one
+      passedMutant$ = passedCheckResult$;
+      checkResult$ = concat(
+        checkResult$,
+        failedCheckResult$.pipe(map((failedMutant) => this.mutationTestReportHelper.reportCheckFailed(failedMutant.mutant, failedMutant.checkResult)))
+      );
     }
-
-    previousPassedMutants$.subscribe({
```

Comment:
using `subscribe` here is not good! Doing that will execute the observable. Subscribing again later will re-execute the observable again. Instead I'm using `tap({ complete: () => {})`.